### ISSUE TITLE
Text

### ISSUE THREAD

**Issue by aVileBroker** [Headstorm/hs-angular-ui#8]
Standard AC:
>Consistent styling across browsers
>0 A11y violations
>90%+ code coverage

Specific API:
* text
  * the text shown in the button
  * no default, optional
  * string
* color
  * the color of the text, icons, spinner, and skeleton
  * default is "charcoal100", optional
  * string
* size
  * the size of the text, icons, spinner, and skeleton
  * default is ".75rem", optional
  * string
* isProcessing
  * if true, shows a spinner to the left of the button text
  * default is false, optional
  * boolean
* isLoading
  * if true, replaces the button text with a skeleton version
  * default is false, optional
  * boolean
* iconPrefix
  * the name of the material design icon to be shown to the 
  * left of the button
  * no default, optional
  * string
* iconSuffix
  * the name of the material design icon to be shown to the 
right of the button
  * no default, optional
  * string
className
className applied to the button container
no default, optional
string
style
inline style object applied to the button container
no default, optional
object
